File: src/main/java/edu/tamu/cap/model/repo/RepositoryViewRepo.java
package edu.tamu.cap.model.repo;

import java.util.List;

import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

import edu.tamu.cap.model.RepositoryView;
import edu.tamu.cap.model.repo.custom.RepositoryViewRepoCustom;
import edu.tamu.weaver.data.model.repo.WeaverRepo;

public interface RepositoryViewRepo extends WeaverRepo<RepositoryView>, RepositoryViewRepoCustom, JpaSpecificationExecutor<RepositoryView> {

	RepositoryView findByName(String name);

	List<RepositoryView> findByRootUriContainingIgnoreCase(String rootUri);

}

File: src/main/webapp/app/repo/repositoryViewRepo.js
cap.repo("RepositoryViewRepo", function (WsApi, api) {
  var repositoryViewRepo = this;

  repositoryViewRepo.scaffold = {
    name: "Labs Fedora",
    rootUri: "https://api-dev.library.tamu.edu/fcrepo/rest/",
    username: "fedoraAdmin",
    password: "secret3",
    type: "",
    schemas: [],
    curators: []
  };

  repositoryViewRepo.getTypes = function (types) {
    var typesPromise = WsApi.fetch(repositoryViewRepo.mapping.getTypes);
    typesPromise.then(function (res) {
      angular.extend(types, angular.fromJson(res.body).payload['ArrayList<HashMap>']);
    });
    return typesPromise;
  };

  repositoryViewRepo.findByName = function (name) {
    var repositoryViews = repositoryViewRepo.getAll();
    for (var i in repositoryViews) {
      var repositoryView = repositoryViews[i];
      if (repositoryView.name === name) {
        return repositoryView;
      }
    }
  };

  repositoryViewRepo.verifyPing = function (repositoryView) {
    return WsApi.fetch(api.VerifyRepositoryViewSettings.verifyPing, {
      pathValues: {
        type: repositoryView.type
      },
      data: repositoryView
    });
  };

  repositoryViewRepo.verifyAuth = function (repositoryView) {
    return WsApi.fetch(api.VerifyRepositoryViewSettings.verifyAuth, {
      pathValues: {
        type: repositoryView.type
      },
      data: repositoryView
    });
  };

  repositoryViewRepo.verifyContent = function (repositoryView) {
    return WsApi.fetch(api.VerifyRepositoryViewSettings.verifyContent, {
      pathValues: {
        type: repositoryView.type
      },
      data: repositoryView
    });
  };

  return repositoryViewRepo;
});
